modular_skyrat/modules/vore/code/obj/belly.dm

```diff
@@ -0,0 +1,304 @@
+					continue
+				prey.apply_damage(4, BURN)
+				if (prey.stat == DEAD)
+					var/pred_message = vore_replace(data[LIST_DIGEST_PRED], owner, prey, name)
+					var/prey_message = vore_replace(data[LIST_DIGEST_PREY], owner, prey, name)
+					vore_message(owner, pred_message, SEE_OTHER_MESSAGES, warning=TRUE)
+					vore_message(prey, prey_message, SEE_OTHER_MESSAGES, warning=TRUE)
+					prey.release_belly_contents()
+					for (var/obj/item/item in prey)
+						if (!prey.dropItemToGround(item))
+							qdel(item)
+					if (prey.check_vore_toggle(LEAVE_ESSENCE_CUBE, VORE_MECHANICS_TOGGLES))
+						new /obj/item/essence_cube(src, prey)
+					else
+						qdel(prey)
+					should_update = TRUE
+				else
+					all_done = FALSE
+
+		//may wanna make this use nutrition in the future or something
+		if (VORE_MODE_ABSORB)
+			for (var/mob/living/prey in absorbed)
+				if (!prey.check_vore_toggle(ABSORBABLE, VORE_MECHANICS_TOGGLES))
+					absorbed -= prey
+					should_update = TRUE
+					continue
+				if (absorbed[prey] < 100)
+					absorbed[prey] += 2
+					all_done = FALSE
+			for (var/mob/living/prey in src)
+				if ((prey in absorbed) || !prey.check_vore_toggle(ABSORBABLE, VORE_MECHANICS_TOGGLES))
+					if (prey in absorbing)
+						absorbing -= prey
+					continue
+				if (!absorbing[prey])
+					absorbing[prey] = 0
+				absorbing[prey] += 2 //fires every 2 seconds, so this will take 100 seconds
+				if (absorbing[prey] >= 100)
+					RegisterSignal(prey, COMSIG_PARENT_EXAMINE, .proc/examine_absorb)
+					var/pred_message = vore_replace(data[LIST_ABSORB_PRED], owner, prey, name)
+					var/prey_message = vore_replace(data[LIST_ABSORB_PREY], owner, prey, name)
+					vore_message(owner, pred_message, SEE_OTHER_MESSAGES, warning=TRUE)
+					vore_message(prey, prey_message, SEE_OTHER_MESSAGES, warning=TRUE)
+					absorbing -= prey
+					absorbed[prey] = 100
+					should_update = TRUE
+				else
+					all_done = FALSE
+
+		if (VORE_MODE_UNABSORB)
+			for (var/mob/living/prey in absorbed)
+				if (!(prey in src))
+					absorbed -= prey
+					should_update = TRUE
+					continue
+				absorbed[prey] -= 2 //fires every 2 seconds, so this will take 100 seconds
+				if (absorbed[prey] <= 0)
+					UnregisterSignal(prey, COMSIG_PARENT_EXAMINE)
+					var/pred_message = vore_replace(data[LIST_UNABSORB_PRED], owner, prey, name)
+					var/prey_message = vore_replace(data[LIST_UNABSORB_PREY], owner, prey, name)
+					vore_message(owner, pred_message, SEE_OTHER_MESSAGES, warning=TRUE)
+					vore_message(prey, prey_message, SEE_OTHER_MESSAGES, warning=TRUE)
+					absorbed -= prey
+					should_update = TRUE
+				else
+					all_done = FALSE
+			for (var/mob/living/prey in absorbing)
+				if (!(prey in src))
+					absorbing -= prey
+					should_update = TRUE
+					continue
+				absorbing[prey] -= 2
+				if (absorbing[prey] <= 0)
+					absorbing -= prey
+				else
+					all_done = FALSE
+
+	if (all_done)
+		STOP_PROCESSING(SSvore, src)
```

Comment:
as mentioned before, move this to normal processing, doesn't need to be on its own subsystem